### DISCUSSION TITLE
Using LOOS in a Jupiter notebook

### DISCUSSION THREAD

**Discussion by krb0073** [GrossfieldLab/loos#38] [Q&A] [answered]
Hey all, 
 I was wondering if anyone has used loos from inside a Jupiter notebook before. Before I switched to only using the conda environment I added that path of loos and pyloos to the notebook environment and was able to have it work. I have have forgotten what I did though.  After the fresh install of loos I can't seem to get it to work. 

Here is what I have tried:

1. Manually adding the path into the script 
2.  I have tried adding the environment(i followed [this guide](https://medium.com/@nrk25693/how-to-add-your-conda-environment-to-your-jupyter-notebook-in-just-4-steps-abeab8b8d084)

Any suggestions would be sweet,
Kyle

**Comment by agrossfield** [comment]
I got it to work, as long as you use a jupyter from the same conda env. I think that’s necessary, because jupyter and jupyterlab assume a specific python.

If you’re already doing that, what error are you getting ?

Dr. Alan Grossfield
Department of Biochemistry and Biophysics
University of Rochester Medical School


On Jan 13, 2021, at 3:39 PM, krb0073 <notifications@github.com> wrote:

﻿

Hey all,
I was wondering if anyone has used loos from inside a Jupiter notebook before. Before I switched to only using the conda environment I added that path of loos and pyloos to the notebook environment and was able to have it work. I have have forgotten what I did though. After the fresh install of loos I can't seem to get it to work.

Here is what I have tried:

  1.  Manually adding the path into the script
  2.  I have tried adding the environment(i followed this guide<https://urldefense.proofpoint.com/v2/url?u=https-3A__medium.com_-40nrk25693_how-2Dto-2Dadd-2Dyour-2Dconda-2Denvironment-2Dto-2Dyour-2Djupyter-2Dnotebook-2Din-2Djust-2D4-2Dsteps-2Dabeab8b8d084&d=DwMCaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=4gqqjxxJbZkZ0NIPhj0Rb6Y2V-ONwZ9x8LxF1koKbaA&s=T6ih6zRBoQnRMU8qwEmEibwqmhj_NgYQ4dj-LdsbsWg&e=>

Any suggestions would beb sweet,
Kyle

**Comment by krb0073** [comment]
when using a notebook in the same conda env. I get an error:
ModuleNotFoundError: No module named 'loos'

I can run python-loos scripts in python just fine which is why i'm confused.
Kyle

**Comment by agrossfield** [comment, marked as answer]
Are you in the conda env in the shell where you launch the notebook? If not, that could be the problem. I wonder if you’re losing the environment in the browser somehow. Alternatively, you could try using jupyterlab, which I just checked the other day, and I know it works.

Cheers,

Alan

________________________________
Dr. Alan Grossfield
Dept of Biochemistry and Biophysics
University of Rochester Medical Center
Phone: 585 276 4193
http://membrane.urmc.rochester.edu
https://orcid.org/0000-0002-5877-2789
Pronouns: He/his

**Comment by krb0073** [comment]
Oh i was not launching the notebook in the conda env. Whoops I was able to get everything working. Thanks for your help.

**Comment by agrossfield** [comment]
I’m glad it turned out to be straightforward— I’ve tripped over similar problems (for instance, when atom relaunched after an update, it’s no longer in the conda env).

Dr. Alan Grossfield
Department of Biochemistry and Biophysics
University of Rochester Medical School


On Jan 13, 2021, at 9:53 PM, krb0073 <notifications@github.com> wrote:

﻿

Oh i was not launching the notebook in the conda env. Whoops I was able to get everything working. Thanks for your help.
